src/html/index.html

```diff
@@ -1,11 +1,36 @@
 <!doctype html>
 <html lang="en">
-  <head>
-    <meta charset="utf-8">
-    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
-    <title><%= htmlWebpackPlugin.options.title || 'Webpack App'%></title>
-  </head>
-  <body>
-    <div id="root"></div>
-  </body>
+
+<head>
+  <%if (htmlWebpackPlugin.options.ipfsHack) { %>
+  <script>
+    (function () {
+      const { origin, pathname } = window.location
+      const ipfsMatch = /.*\/Qm\w{44}/.exec(pathname)
+      const base = document.createElement('base')
+      let baseHref
+      if (ipfsMatch) {
+
+        baseHref = origin + ipfsMatch[0]
+        if (!baseHref.endsWith('/')) baseHref += '/'
+
+        window.BASE_URL = ipfsMatch[0]
```

Comment:
Do we really need the global? can we use the `<base /> href` instead?

Reply:
`base.href` is same as `document.baseURI`, it prepends origin. We'd have to recalculate anyway